common/network-common/src/main/java/org/apache/spark/network/TransportContext.java

```diff
@@ -123,8 +123,7 @@ public TransportContext(
     this.rpcHandler = rpcHandler;
     this.closeIdleConnections = closeIdleConnections;
 
-    if (conf.getModuleName() != null &&
-        conf.getModuleName().equalsIgnoreCase("shuffle") &&
+    if ("shuffle".equalsIgnoreCase(conf.getModuleName()) &&
```

Comment:
Previously, there is no NPE in this case. So, this is just a code-implication instead of avoiding any NPE.